### PR TITLE
DOCS-6927 - Data Jobs Monitoring

### PR DESCRIPTION
**Pull request by cswatt**
### What does this PR do? What is the motivation?

Adds documentation for Data Jobs Monitoring.

### Merge instructions

We are, for now, using the same left nav icon as Data Streams Monitoring.

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/data_jobs/databricks.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+Follow these steps to enable Data Jobs Monitoring for Databricks.
+
+1. [Configure the Datadog-Databricks integration](#configure-the-datadog-databricks-integration) with your Databricks API token.
+1. [Install the Datadog Agent](#install-the-datadog-agent-on-your-databricks-clusters) on your Databricks cluster(s).
+1. [Add your Datadog API key](#add-your-datadog-api-key-in-databricks) in Databricks.
+
+### Configure the Datadog-Databricks integration 
+
+1. In your Databricks workspace, click **User Settings**. Then click on **Access Tokens**.
```
Comment (paul-laffon-dd) [inline content/en/data_jobs/databricks.md:26]: Databricks recently slightly changed their UI

The new path for Access tokens is: User Settings -> Developer Tab -> Manage Access tokens

image

File: content/en/data_jobs/databricks.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+1. At the bottom of the page, go to the **Init Scripts** tab.
+   {{< img src="data_jobs/databricks/init_scripts.png" alt="Databricks UI, cluster configuration advanced options,  Init Scripts tab. A 'Destination' drop-down and an 'Init script path' file selector." style="width:80%;" >}}
+   - Under the **Destination** drop-down, select `Workspace`.
+   - Under **Init script path**, enter the path to your init script.
+   - Click **Add**.
+
+[1]: /resources/sh/data_jobs/datadog_databricks_job_monitoring_init_without_logs.sh
+{{% /tab %}}
+{{% tab "Global" %}}
+1. In Databricks, go to **Admin Settings** and click the **Global Init Scripts** tab.
```
Comment (paul-laffon-dd) [inline content/en/data_jobs/databricks.md:51]: This Databricks UI also recently changed

The path is now: Admin Settings -> Compute -> Manage Global init scripts

image

File: content/en/data_jobs/_index.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+- Track data processing jobs across your accounts and workspaces, and see which take up the most compute resources or have inefficiencies.
+- Receive an alert when a job fails—or when a job that should be complete is still running.
+- Analyze job execution details and stack traces.
+- Correlate infrastructure metrics, Spark metrics from the Spark UI, logs, and cluster configuration.
+- Compare multiple runs to facilitate troubleshooting, and to optimize provisioning and configuration during deployment.
+
+## Setup
+
+Data Jobs Monitoring is supported for Amazon EMR, Databricks, and Spark on Kubernetes.
```
Comment (warrierr) [inline content/en/data_jobs/_index.md:27]:
```suggestion
Data Jobs Monitoring is supported for Amazon EMR, Databricks (AWS, Azure, Google Cloud), and Spark on Kubernetes.
```

File: content/en/data_jobs/_index.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+{{< partial name="data_jobs/setup-platforms.html" >}}
+
+<br/>
+
+## Explore Data Jobs Monitoring
+
+### Take inventory of your data processing jobs
+
+View all jobs across cloud accounts and workspaces. See CPU allocation, cost estimation, and other metrics to identify opportunities for optimization.
```
Comment (warrierr) [inline content/en/data_jobs/_index.md:39]:
```suggestion
View all jobs across cloud accounts and workspaces. Identify failing jobs to take action on, or those with high idle CPU that are using a lot of compute that should be optimized.
```

File: content/en/data_jobs/databricks.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+1. Click **Generate new token**, enter a comment, and click **Generate**. Take note of your token.
+1. In Datadog, open the Databricks integration tile.
+1. On the **Configure** tab, click **Add New**. 
+1. Enter a workspace name, your Databricks workspace URL, and the Databricks token you generated.
+   {{< img src="data_jobs/databricks/configure-token.png" alt="In the Datadog-Databricks integration tile, a Databricks workspace is displayed. This workspace has a name, URL, and API token." style="width:100%;" >}}
+
+
+### Install the Datadog Agent on your Databricks cluster(s)
+
+Use [this init script][2] to install the Datadog Agent globally, or on a specific Databricks cluster.
```
Comment (warrierr) [inline content/en/data_jobs/databricks.md:36]: Can we give the customer this script as the deafault one that collects logs by default: https://drive.google.com/file/d/19Z50fVzeoWbHU4L4XwPJspGmOyx_UX3R/view?usp=drive_link that does include 

For the script you are currently linking to, we can show that one as an option if the customer doesn't want logs collected.

**Review by warrierr — CHANGES_REQUESTED** [review]
Changes to the overview page and one request for databricks setup

File: content/en/data_jobs/_index.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+### Analyze and troubleshoot individual jobs
+
+Click on a job to see how it performed across multiple runs, as well as error messages for failed runs.
+
+{{< img src="data_jobs/djm_job.png" alt="Job list - CPU breakdown" style="width:100%;" >}}
+
+### View details of each individual run
+
+Clicking on a run opens a side panel with details of how much time was spent on each Spark job and stage, helping you to understand the breakdown of resource consumption and Spark metrics. From this panel, you can also view executor and driver node resource utilization, as well as job and cluster configuration.
```
[the remark this answers is not in the document]
Comment (cswatt) [reply content/en/data_jobs/_index.md:51]:
```suggestion
Clicking on a run opens a side panel with details of how much time was spent on each Spark job and stage, along with a breakdown of resource consumption and Spark metrics, such as idle executor CPU, input/output data volume, shuffling, and disk spill. From this panel, you can correlate the execution with executor and driver node resource utilization, logs, and the job and cluster configuration. 
```

File: content/en/data_jobs/_index.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+To get started, select your platform and follow the installation instructions:
+
+{{< partial name="data_jobs/setup-platforms.html" >}}
+
+<br/>
+
+## Explore Data Jobs Monitoring
+
+### Take inventory of your data processing jobs
```
[the remark this answers is not in the document]
Comment (cswatt) [reply content/en/data_jobs/_index.md:37]:
```suggestion
### Easily identify unreliable and inefficient jobs
```

File: content/en/data_jobs/_index.md (hunk 4 above)
Comment (cswatt) [reply content/en/data_jobs/_index.md:39]: View all jobs across cloud accounts and workspaces. Identify failing jobs to take action on, or find jobs with high idle CPU that are using a lot of compute and should be optimized.

File: content/en/data_jobs/_index.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+View all jobs across cloud accounts and workspaces. See CPU allocation, cost estimation, and other metrics to identify opportunities for optimization.
+
+{{< img src="data_jobs/djm_demolist.png" alt="Job list - CPU breakdown" style="width:100%;" >}}
+
```
[the remark this answers is not in the document]
Comment (cswatt) [reply content/en/data_jobs/_index.md:42]: added below; altered copy to remove marketing language

File: content/en/data_jobs/_index.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+{{< img src="data_jobs/djm_job.png" alt="Job list - CPU breakdown" style="width:100%;" >}}
+
+### View details of each individual run
+
+Clicking on a run opens a side panel with details of how much time was spent on each Spark job and stage, helping you to understand the breakdown of resource consumption and Spark metrics. From this panel, you can also view executor and driver node resource utilization, as well as job and cluster configuration. 
+
+{{< img src="data_jobs/djm_run.png" alt="Job list - CPU breakdown" style="width:100%;" >}}
+
+For a failed run, you can see the stack trace to determine where and how this failure occurred.
```
[the remark this answers is not in the document]
Comment (cswatt) [reply content/en/data_jobs/_index.md:65]: Added above, with copy edits to remove marketing language. I think it's important to conceptualize that to view spark task metrics, it's another tab on the individual run details panel. For that reason, I don't think it should have its own heading.

**Review by warrierr — APPROVED** [review]

File: static/resources/sh/data_jobs/datadog_databricks_job_monitoring_init.sh [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+cat <<EOF > /tmp/start_datadog.sh
+
+#!/bin/bash
+
+date -u +"%Y-%m-%d %H:%M:%S UTC"
+echo "Running on the driver? $DB_IS_DRIVER"
+echo "Driver ip: $DB_DRIVER_IP"
+
+DB_CLUSTER_NAME=$(echo "$DB_CLUSTER_NAME" | sed -e 's/ /_/g' -e "s/'/_/g" -e 's/"/_/g')
```
Comment (paul-laffon-dd) [inline static/resources/sh/data_jobs/datadog_databricks_job_monitoring_init.sh:18]:
```suggestion
DB_CLUSTER_NAME="$(echo "$DB_CLUSTER_NAME" | sed 's/[^a-zA-Z0-9_:.-]/_/g')"
```

File: static/resources/sh/data_jobs/datadog_databricks_job_monitoring_init.sh [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+DD_TAGS="environment:\${DD_ENV}","databricks_cluster_id:\${DB_CLUSTER_ID}","databricks_cluster_name:\${DB_CLUSTER_NAME}","databricks_container_ip:\${DB_CONTAINER_IP}","databricks_is_driver:\${DB_IS_DRIVER}","databricks_instance_type:\${DB_INSTANCE_TYPE}","databricks_is_job_cluster:\${DB_IS_JOB_CLUSTER}","data_workload_monitoring_trial:true"
+
+if [[ $DB_CLUSTER_NAME =~ ^job-([0-9]+)- ]]; then
```
Comment (paul-laffon-dd) [inline static/resources/sh/data_jobs/datadog_databricks_job_monitoring_init.sh:21]:
```suggestion
if [[ \${DB_CLUSTER_NAME} =~ ^job-([0-9]+)- ]]; then
```

File: static/resources/sh/data_jobs/datadog_databricks_job_monitoring_init_without_logs.sh [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+cat <<EOF > /tmp/start_datadog.sh
+
+#!/bin/bash
+
+date -u +"%Y-%m-%d %H:%M:%S UTC"
+echo "Running on the driver? $DB_IS_DRIVER"
+echo "Driver ip: $DB_DRIVER_IP"
+
+DB_CLUSTER_NAME=$(echo "$DB_CLUSTER_NAME" | sed -e 's/ /_/g' -e "s/'/_/g" -e 's/"/_/g')
```
Comment (paul-laffon-dd) [inline static/resources/sh/data_jobs/datadog_databricks_job_monitoring_init_without_logs.sh:18]:
```suggestion
DB_CLUSTER_NAME="$(echo "$DB_CLUSTER_NAME" | sed 's/[^a-zA-Z0-9_:.-]/_/g')"
```

File: static/resources/sh/data_jobs/datadog_databricks_job_monitoring_init_without_logs.sh [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+DD_TAGS="environment:\${DD_ENV}","databricks_cluster_id:\${DB_CLUSTER_ID}","databricks_cluster_name:\${DB_CLUSTER_NAME}","databricks_container_ip:\${DB_CONTAINER_IP}","databricks_is_driver:\${DB_IS_DRIVER}","databricks_instance_type:\${DB_INSTANCE_TYPE}","databricks_is_job_cluster:\${DB_IS_JOB_CLUSTER}","data_workload_monitoring_trial:true"
+
+if [[ $DB_CLUSTER_NAME =~ ^job-([0-9]+)- ]]; then
```
Comment (paul-laffon-dd) [inline static/resources/sh/data_jobs/datadog_databricks_job_monitoring_init_without_logs.sh:21]:
```suggestion
if [[ \${DB_CLUSTER_NAME} =~ ^job-([0-9]+)- ]]; then
```

**Review by paul-laffon-dd — COMMENTED** [review]
Added suggestions on the databricks init script, because the current version is failing if special characters are present in the `DB_CLUSTER_NAME`

File: layouts/shortcodes/djm-runtime-tagging.md [hunk 14]
```diff
@@ -0,0 +1,14 @@
+You can set tags on Spark spans at runtime. These tag are applied _only_ to spans that start after the tag is added.
```
Comment (hestonhoffman) [inline layouts/shortcodes/djm-runtime-tagging.md:1]:
```suggestion
You can set tags on Spark spans at runtime. These tags are applied _only_ to spans that start after the tag is added.
```

File: content/en/data_jobs/databricks.md [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+## Further Reading
+
+{{< partial name="whats-next/whats-next.html" >}}
+
+[1]: https://app.datadoghq.com/integrations/databricks?search=databricks
+[2]: /resources/sh/data_jobs/datadog_databricks_job_monitoring_init_without_logs.sh
+[3]: https://app.datadoghq.com/organization-settings/api-keys
+[4]: https://docs.databricks.com/en/security/secrets/index.html
+[5]: /getting_started/site/
+[6]: https://app.datadoghq.com/apm/data-jobs
```
Comment (hestonhoffman) [inline content/en/data_jobs/databricks.md:101]:
```suggestion
[6]: https://app.datadoghq.com/data-jobs/
```

File: content/en/data_jobs/kubernetes.md [hunk 16]
```diff
@@ -0,0 +1,213 @@
...
+
+## Further Reading
+
+{{< partial name="whats-next/whats-next.html" >}}
+
+[1]: /containers/kubernetes/installation/?tab=operator
+[2]: /containers/cluster_agent/admission_controller/?tab=operator
+[3]: /containers/datadog_operator
+[4]: https://helm.sh
+[5]: https://app.datadoghq.com/apm/data-jobs
```
Comment (hestonhoffman) [inline content/en/data_jobs/kubernetes.md:212]:
```suggestion
[5]: https://app.datadoghq.com/data-jobs/
```

File: content/en/data_jobs/emr.md [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+{{< partial name="whats-next/whats-next.html" >}}
+
+[1]: https://app.datadoghq.com/organization-settings/api-keys
+[2]: https://console.aws.amazon.com/secretsmanager/
+[3]: https://docs.aws.amazon.com/secretsmanager/latest/userguide/rotating-secrets.html
+[4]: https://console.aws.amazon.com/emr
+[5]: https://console.aws.amazon.com/iam/
+[6]: /resources/sh/data_jobs/datadog_emr_job_monitoring_init_v2.sh
+[7]: /getting_started/site/
+[8]: https://app.datadoghq.com/apm/data-jobs
```
Comment (hestonhoffman) [inline content/en/data_jobs/emr.md:107]:
```suggestion
[8]: https://app.datadoghq.com/data-jobs/
```

**Review by hestonhoffman — APPROVED** [review]
Couple minor edits, but this looks great! I'm not going to merge cos there are some outstanding regex changes to the scripts.

**Pull request merged**